### ISSUE TITLE
Invalid certificates - breaks content negotiation (and also the blog page)

### ISSUE THREAD

**Issue by markwilkinson** [dydra/support#56]
content negotiation on my landing page redirects me to rdf.dydra.com...  Seems that the certs are invalid:

$ curl -L -v -H "Accept: text/turtle" https://rdf.dydra.com/markw/polyadenylation-sites-in-magnaporthe-oryza.ttl
*   Trying 52.216.240.44...
* TCP_NODELAY set
* Connected to rdf.dydra.com (52.216.240.44) port 443 (#0)
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /home/markw/anaconda3/ssl/cacert.pem
  CApath: none
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-SHA
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: C=US; ST=Washington; L=Seattle; O=Amazon.com Inc.; CN=*.s3.amazonaws.com
*  start date: Nov  7 00:00:00 2018 GMT
*  expire date: Feb  7 12:00:00 2020 GMT
*  subjectAltName does not match rdf.dydra.com
* SSL: no alternative certificate subject name matches target host name 'rdf.dydra.com'
* Closing connection 0
* TLSv1.2 (OUT), TLS alert, Client hello (1):
curl: (51) SSL: no alternative certificate subject name matches target host name 'rdf.dydra.com'

**Comment by lisp** [comment]
yes, that certificate is not ours.
i am not clear how you get to a request from rdf.dydra.com.
i would expect this to be the usage pattern:

```
$  curl -L -s -H "Accept: text/turtle" https://dydra.com/markw/polyadenylation-sites-in-magnaporthe-oryza.ttl | rapper --count -i turtle /dev/stdin
rapper: Parsing URI file:///dev/stdin with parser turtle
rapper: Parsing returned 510508 triples

```

**Comment by markwilkinson** [comment]
It's coming from content negotiation on my Dydra landing page, here:

$ curl -L -v -H "Accept: text/turtle" https://dydra.com/markw/polyadenylation-sites-in-magnaporthe-oryza
*   Trying 94.130.163.224...
* TCP_NODELAY set
* Connected to dydra.com (94.130.163.224) port 443 (#0)
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /home/markw/anaconda3/ssl/cacert.pem
  CApath: none
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: OU=Domain Control Validated; OU=PositiveSSL Wildcard; CN=*.dydra.com
*  start date: Jul  4 00:00:00 2018 GMT
*  expire date: Jul 12 23:59:59 2020 GMT
*  subjectAltName: host "dydra.com" matched cert's "dydra.com"
*  issuer: C=GB; ST=Greater Manchester; L=Salford; O=COMODO CA Limited; CN=COMODO RSA Domain Validation Secure Server CA
*  SSL certificate verify ok.
> GET /markw/polyadenylation-sites-in-magnaporthe-oryza HTTP/1.1
> Host: dydra.com
> User-Agent: curl/7.60.0
> Accept: text/turtle
> 
< HTTP/1.1 302 Found
< Server: nginx/1.10.3 (Ubuntu)
< Date: Tue, 24 Sep 2019 10:42:34 GMT
< Content-Type: text/turtle; charset=utf-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< Status: 302 Found
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Credentials: true
< Access-Control-Allow-Headers: Authorization, Content-Type, X-Requested-With
< Location: https://rdf.dydra.com/markw/polyadenylation-sites-in-magnaporthe-oryza.ttl
< Cache-Control: no-cache
< X-UA-Compatible: IE=Edge,chrome=1
< X-Runtime: 0.014955
< 
* Ignoring the response-body
* Connection #0 to host dydra.com left intact
* Issue another request to this URL: 'https://rdf.dydra.com/markw/polyadenylation-sites-in-magnaporthe-oryza.ttl'
*   Trying 52.216.146.139...
* TCP_NODELAY set
* Connected to rdf.dydra.com (52.216.146.139) port 443 (#1)
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /home/markw/anaconda3/ssl/cacert.pem
  CApath: none
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-SHA
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: C=US; ST=Washington; L=Seattle; O=Amazon.com Inc.; CN=*.s3.amazonaws.com
*  start date: Nov  7 00:00:00 2018 GMT
*  expire date: Feb  7 12:00:00 2020 GMT
*  subjectAltName does not match rdf.dydra.com
* SSL: no alternative certificate subject name matches target host name 'rdf.dydra.com'
* Closing connection 1
* TLSv1.2 (OUT), TLS alert, Client hello (1):
curl: (51) SSL: no alternative certificate subject name matches target host name 'rdf.dydra.com'

**Comment by markwilkinson** [comment]
What I was actually trying to retrieve was the metadata about my repository, in Turtle, not the repository itself (which is why I was doing content negotiation on the landing page - to get the Turtle representation of the landing page...)

**Comment by lisp** [comment]
the request which you made becomes a graph store protocol request, which yields the content.
i will look into why the "rdf." host is still the redirect target.

if you mean metadata as in the service description, you can retrieve that wth a sparql request with no content.

**Comment by markwilkinson** [comment]
Nope, I don't mean the service description.  I mean the metadata about the deposit.  It's license, etc.  In some structured format.

I think returning me the content of the repository via content negotiation on the landing page is "semantically incorrect" - the landing page is metadata about the repo, so I am asking for a structured representation of that landing page metadata, not for the data itself...

**Comment by lisp** [comment]
> I think returning me the content of the repository via content negotiation on the landing page is "semantically incorrect" - the landing page is metadata about the repo, ...

why?

if you want to work with the metadata, use your "system" repository:

```
$ curl --user "${TOKEN}:" https://dydra.com/james/system.ttl | rapper --count -i turtle /dev/stdin
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0rapper: Parsing URI file:///dev/stdin with parser turtle
100 50493    0 50493    0     0  88739      0 --:--:-- --:--:-- --:--:-- 88584
rapper: Parsing returned 905 triples
```

at one point we supported also resources of the form `account/<account-name>/repository/<repository-name>, but moved to the system repository as it is more general.

**Comment by markwilkinson** [comment]
A few observations:

1) https://dydra.com/markw/system.ttl   returns 404
2) There is no reference to this "system.ttl" in the homepage, so how would a machine know it exists in order to explore it?
3) content negotiation is a request for different representations of the URI being called.  In my case, I am calling content negotiation on the URI of my landing page, not the URI of my dataset... so I expect a structured representation of that landing page, hopefully containing a pointer to the dataset (e.g. as a dcat:Distribution), but not the dataset itself.  That's what I mean by "semantically incorrect" behavior.

**Comment by lisp** [comment]
> 1. https://dydra.com/markw/system.ttl   returns 404

please try it with the sparql or graph store protocol endpoint, as in

    https://dydra.com/markw/system/service

> 2. There is no reference to this "system.ttl" in the homepage, so how would a machine know it exists in order to explore it?

yes, you are correct that we do not expose it actively.
it is appears in the documentation for the access authorization system:

    https://api.dydra.com/authorization/access-control-graph.html?highlight=system

> 3. content negotiation is a request for different representations of the URI being called.  In my case, I am calling content negotiation on the URI of my landing page,  not the URI of my dataset... so I expect a structured representation of that landing page, 


which i would understand to intend to return metadata about that page or that same content encoded as turtle.
if you expect the latter, you have correctly observed that we do not provide that response.

> hopefully containing a pointer to the dataset (e.g. as a dcat:Distribution), but not the dataset itself.  That's what I mean by "semantically incorrect" behavior.

are you looking for the information which is not present in the service description?

**Comment by markwilkinson** [comment]
Yes, I am looking for a machine-readable version of the metadata on the landing page (e.g. title, license, description of the dataset, link to homepage, authorship info,... the usual dublin core stuff that appears as HTML on the current landing page.  This would make the landing page (more) compliant with the FAIR Principles for data publication.

**Comment by lisp** [comment]
as that pages, as you see it, evolved opportunistically in response to customer capability requests, it is not likely, that we would attempt to reconcile it to any standard schema. were a need to arise to conform to "fair principles", we would likely define a sparql view on the system repository and redirect to that.

in addition to which that page composes several independent resource.
the notion, that it is a monolithic resource to be represented in media types other than html is less stable than would serve well as the basis for a ui api and, as, as it were, its composition and constitution are becoming even less stable in the oncoming version, it will be more difficult to reduce it to some predictable turtle document form.

**Comment by markwilkinson** [comment]
Understood.  No worries.  My main question was whether it had the capability now, and I just wasn't seeing it.  If it doesn't, then no problem!  :-)
Cheers!

**Comment by lisp** [comment]
in the future, any such "capability" will be less so than you would like.
the user will decide what appears on a repository web-ui page and the only stable source for metadata related to the repository will be its description in the system repository.
